docs/topic-guides/blendable-web-apps/uniform-design-system.md

```diff
@@ -0,0 +1,90 @@
+
+The names of the CSS custom properties, CSS classes, and custom elements begin with a `uni-` prefix.
+
+Our [KitchenSink](https://kitchensink.starcounter.io/) app demonstrates the use of Uniform design system as well as commonly used 3rd party custom elements that we can endorse, such as a Google map or a combo box.
+
+### Theme configuration through CSS custom properties
+
+Uniform design system comes with a modern-looking default theme.
+
+As a solution owner, you can customize it by providing new values for the [CSS custom properties](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_variables). This allows changing the fonts, colors, etc. without replacing the stylesheet.
+
+You can provide new values directly in a stylesheet in a custom composition. It requires some skill in CSS and HTML but allows for spectacular results such as providing a theme configuration only for a specific part of the app.
+
+If you don't need that much of control and prefer a simple to use GUI, use the "Theme configurations" feature of [Blending](https://github.com/Starcounter/Blending) app suite. It is an administration tool that stores the values of CSS custom properties in the database. It can provide theme configurations for specific layouts that affect all Uniform components on the screen.
+
+{% hint style="warning" %} Attaching CSS custom properties through "Theme configurations" sets them on the global scope - the `body` of the light DOM (https://github.com/Starcounter/Blending/issues/232). Providing CSS custom properties values in custom compositions attaches them in shadow DOM to the scope that you choose. {% endhint %}
+
+{% hint style="info" %} For more sophisticated theming we would love to use native [CSS shadow parts and themes](https://meowni.ca/posts/part-theme-explainer/). Unfortunately, it's not supported yet by any browser. {% endhint %}
+
+### Extend or replace the theme with HTML, CSS and JavaScript
+
+The architecture of Uniform design system is based on the inversion of control. This means that all parts of it can be extended or replaced by the solution owner when needed. It allows you to completely change the look or behavior without changing the app source code.
+
+As a solution owner, you are free to add, enhance, replace or hack the presentation layer using your HTML, CSS or JavaScript code. The only technical limitation is Shadow DOM encapsulation. 
+
+For example, when all apps consistently use Uniform components, all the date pickers are presented using `<uni-date-picker>`. You can drastically change its look or behavior by creating your own custom element.
```

Comment:
I think that method is unrelated to shadow dom composition (previous paragraph) and you would not do that in Shadow DOM, you do that by replacing files on server side and works thanks to the fact that apps follow the convention of Uni-custom elements.

So this is not "extending and replacing theme/presentation with HTML, CSS and JavaScript", it's rather "extending and replacing HTML, CSS and JavaScript theme/presentation with your OS file manager and running new app". Those, in my opinion, should not be treated as one.

The feature that would be related to the previous paragraph is to replace the single instance of `uni-date-picker` with `other-party-date-picker`.